pacrunner.go

```diff
@@ -73,9 +73,14 @@ func (pr *PACRunner) Update(pacjs []byte) error {
 	return nil
 }
 
-func (pr *PACRunner) FindProxyForURL(u *url.URL) (string, error) {
+func (pr *PACRunner) FindProxyForURL(u url.URL) (string, error) {
 	pr.Lock()
 	defer pr.Unlock()
+	if u.Scheme == "" {
+		// When a net/http Server parses a CONNECT request, the URL will
+		// have no Scheme. In that case, assume the scheme is "https".
+		u.Scheme = "https"
```

Comment:
This could be a little smarter by looking at the port if specified. All the CONNECT examples specify it, but a quick reading of the spec suggests that maybe port is not mandatory. But if port is 80, you could probably safely assume http and not https.

Reply:
I was considering that, but decided against it because I didn't want to have to maintain a mapping. It would be nice if we could use something like [this services map](https://github.com/golang/go/blob/35a588109b2a6d8b610be08d32aaf99ef1549085/src/net/lookup.go#L36-L55), which is exposed in [net.LookupPort()](https://github.com/golang/go/blob/35a588109b2a6d8b610be08d32aaf99ef1549085/src/net/lookup.go#L406), although that mapping is going in the opposite direction to what we'd need here.

Also iiuc port 80 could mean ws:// too, and maybe that's more likely to be the case when someone does a CONNECT request to port 80?

Reply:
Also this: https://gist.github.com/mahmoud/2fe281a8daaff26cfe9c15d2c5bf5c8b

Reply:
There is `/etc/services` too.